src/Buildalyzer/Caching/DotnetInfoCache.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace Buildalyzer.Caching
+{
+    internal static class DotnetInfoCache
+    {
+        private static Dictionary<string, List<string>> cache = new Dictionary<string, List<string>>();
+
+        [Pure]
+        public static List<string> GetCache(string path)
+        {
+            return cache.FirstOrDefault(x => x.Key == path).Value;
+        }
+
+        [Pure]
```

Comment:
This is not a pure method.

Reply:
@Corniel, I really believe that neither of the 2 would be Pure, correct? as a requirement would only be local variables.

Reply:
The question is always: If I do not use the result of the call, is the call still useful? Of not, the method is pure.